contracts/interfaces/IEnigma.sol

```diff
@@ -9,4 +9,5 @@ interface IEnigma {
     function getSecretContractAddresses(uint _start, uint _stop) external view returns (bytes32[] memory);
     function getAllSecretContractAddresses() external view returns (bytes32[] memory);
     function getSigningAddress() external view returns (address);
+    function transferWorkerStakePostUpgrade(address _workerAddress, address _signer, bytes calldata _sig) external returns (uint);
```

Comment:
Why have you chosen to include here `transferWorkerStakePostUpgrade()` but not `upgradeEnigmaContractImpl()` ?

Reply:
I added only `transferWorkerStakePostUpgrade` since I was using it in the mock contract registration function, and wasn't using `upgradeEnigmaContractImpl` in the context of the interface anywhere, so didn't include it.